src/schemas.jl

```diff
@@ -478,18 +478,42 @@ function _generate_record_type_definitions(schema_version::SchemaVersion, record
     field_definitions = Expr[]
     field_assignments = Expr[]
     for (fname, ftype) in pairs(record_fields)
+        fsym = Base.Meta.quot(fname)
         T = Base.Meta.quot(ftype)
         fdef = :($fname::$T)
         info = get(declared_field_infos, fname, nothing)
         if !isnothing(info)
+            fcatch = quote
+                if $fname isa $(info.type)
+                    throw(ArgumentError("Invalid value set for field $($fsym) ($(repr($(fname))))"))
```

Comment:
maybe put the field names in backticks so they stand out?

```suggestion
                    throw(ArgumentError("Invalid value set for field `$($fsym)` ($(repr($(fname))))"))
```